File: codegen/results/v002/haiku/20260411_153257/python/005_unit_test_writer/solution.py
def calculate_discount(price: float, quantity: int) -> float:
    if quantity < 10:
        return price * quantity
    elif quantity < 50:
        return price * quantity * 0.9
    else:
        return price * quantity * 0.75


def run_test(description: str, result: float, expected: float) -> None:
    if abs(result - expected) < 1e-9:
        print(f"PASS: {description}")
    else:
        print(f"FAIL: {description} (expected: {expected}, got: {result})")


tests = [
    ("quantity 9 - no discount",       calculate_discount(10.0, 9),  10.0 * 9),
    ("quantity 10 - 10% discount",     calculate_discount(10.0, 10), 10.0 * 10 * 0.9),
    ("quantity 49 - 10% discount",     calculate_discount(10.0, 49), 10.0 * 49 * 0.9),
    ("quantity 50 - 25% discount",     calculate_discount(10.0, 50), 10.0 * 50 * 0.75),
    ("quantity 1 - no discount",       calculate_discount(5.0, 1),   5.0 * 1),
    ("quantity 100 - 25% discount",    calculate_discount(20.0, 100), 20.0 * 100 * 0.75),
]

for description, result, expected in tests:
    run_test(description, result, expected)

File: codegen/results/v002/haiku/20260411_153257/typescript/005_unit_test_writer/solution.ts
function calculateDiscount(price: number, quantity: number): number {
    if (quantity < 10) return price * quantity;
    else if (quantity < 50) return price * quantity * 0.9;
    else return price * quantity * 0.75;
}

interface TestCase {
    description: string;
    price: number;
    quantity: number;
    expected: number;
}

function runTest(tc: TestCase): void {
    const got = calculateDiscount(tc.price, tc.quantity);
    if (Math.abs(got - tc.expected) < 0.0001) {
        console.log(`PASS: ${tc.description}`);
    } else {
        console.log(`FAIL: ${tc.description} (expected: ${tc.expected}, got: ${got})`);
    }
}

const tests: TestCase[] = [
    { description: "quantity 9 (below 10, no discount)", price: 10, quantity: 9, expected: 90 },
    { description: "quantity 10 (boundary, 10% discount)", price: 10, quantity: 10, expected: 90 },
    { description: "quantity 49 (below 50, 10% discount)", price: 10, quantity: 49, expected: 441 },
    { description: "quantity 50 (boundary, 25% discount)", price: 10, quantity: 50, expected: 375 },
    { description: "quantity 1 (single item, no discount)", price: 20, quantity: 1, expected: 20 },
    { description: "quantity 100 (large order, 25% discount)", price: 5, quantity: 100, expected: 375 },
];

for (const tc of tests) {
    runTest(tc);
}
